code/registration.py

```diff
@@ -1549,45 +1746,62 @@ def singleplane_motion_correction(
         "steps=1.",
     )
 
-    # Parse command-line arguments
-    args = parser.parse_args()
+    return parser.parse_args()
+
+
+if __name__ == "__main__":  # pragma: nocover
+    # Set the log level and name the logger
+    logger = logging.getLogger("Suite2P motion correction")
+    logger.setLevel(logging.INFO)
+
+    # Create an ArgumentParser object
+    parser = parse_arguments()
     # General settings
-    output_dir = Path(args.output_dir)
-    data_dir = Path("../data")
+    data_dir = Path(parser.input)
+    output_dir = Path(parser.output_dir)
     session_fp = next(data_dir.rglob("session.json"))
     description_fp = next(data_dir.rglob("data_description.json"))
+    tmp_dir = Path(parser.tmp_dir)
     with open(session_fp, "r") as j:
         session = json.load(j)
     with open(description_fp, "r") as j:
         data_description = json.load(j)
-    for i in session["data_streams"]:
-        frame_rate_hz = [j["frame_rate"] for j in i["ophys_fovs"]]
-        if frame_rate_hz:
-            break
-    frame_rate_hz = frame_rate_hz[0]
-    if isinstance(frame_rate_hz, str):
-        frame_rate_hz = float(frame_rate_hz)
-    reference_image_fp = ""
+    frame_rate_hz = get_frame_rate(session)
     unique_id = "_".join(str(data_description["name"]).split("_")[-3:])
-    if "Bergamo" in session["rig_id"]:
-        h5_file, output_dir, reference_image_fp = singleplane_motion_correction(
-            data_dir, output_dir, session, unique_id, debug=args.debug
-        )
-    else:
-        h5_file, output_dir, frame_rate_hz = multiplane_motion_correction(
-            data_dir, output_dir, debug=args.debug
-        )
+    reference_image_fp = ""
 
+    if parser.data_type == "TIFF":
+        input_file = [i for i in data_dir.rglob("pophys/*.tif*")] 
+    else:
+        if "Bergamo" in session.get("rig_id", ""):
+            h5_file, output_dir, reference_image_fp = singleplane_motion_correction(
+                data_dir, output_dir, session, unique_id, debug=parser.debug
+            )
+        else:
+            h5_file, output_dir, frame_rate_hz = multiplane_motion_correction(
+                data_dir, output_dir, debug=parser.debug
+            )
+        input_file = str(h5_file)
     # We convert to dictionary
-    args = vars(args)
-    h5_file = str(h5_file)
+    args = vars(parser)
+    if not frame_rate_hz:
+        frame_rate_hz = parser.frame_rate
+        logging.warning("Using default frame rate of 31.0 Hz")
     reference_image = None
-    meta_jsons = list(data_dir.glob("*/*.json"))
     args["refImg"] = []
     if reference_image_fp:
         args["refImg"] = [reference_image_fp]
 
     # We construct the paths to the outputs
+    if isinstance(input_file, list):
+        logging.info("Processing tiff timeseries")
+        basename = unique_id
+        data = tiff_to_numpy(input_file)
+        h5_tmp = str(tmp_dir / f"{basename}.h5")
+        array_to_h5(data, h5_tmp)
```

Comment:
Do you really have to convert the list of tiffs to an h5? I thought suite2p can handle the former directly as well